keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/artifact/ImageHandler.kt

```diff
@@ -0,0 +1,158 @@
+package com.netflix.spinnaker.keel.bakery.artifact
+
+import com.netflix.frigga.ami.AppVersion
+import com.netflix.spinnaker.igor.ArtifactService
+import com.netflix.spinnaker.keel.actuation.ArtifactHandler
+import com.netflix.spinnaker.keel.api.actuation.Task
+import com.netflix.spinnaker.keel.api.actuation.TaskLauncher
+import com.netflix.spinnaker.keel.api.artifacts.DebianArtifact
+import com.netflix.spinnaker.keel.api.artifacts.DeliveryArtifact
+import com.netflix.spinnaker.keel.bakery.BaseImageCache
+import com.netflix.spinnaker.keel.clouddriver.ImageService
+import com.netflix.spinnaker.keel.core.NoKnownArtifactVersions
+import com.netflix.spinnaker.keel.events.ArtifactRegisteredEvent
+import com.netflix.spinnaker.keel.model.Job
+import com.netflix.spinnaker.keel.persistence.KeelRepository
+import com.netflix.spinnaker.keel.persistence.NoSuchArtifactException
+import com.netflix.spinnaker.keel.telemetry.ArtifactCheckSkipped
+import org.slf4j.LoggerFactory
+import org.springframework.context.ApplicationEventPublisher
+
+class ImageHandler(
+  private val repository: KeelRepository,
+  private val baseImageCache: BaseImageCache,
+  private val igorService: ArtifactService,
+  private val imageService: ImageService,
+  private val publisher: ApplicationEventPublisher,
+  private val taskLauncher: TaskLauncher,
+  private val defaultCredentials: BakeCredentials
+) : ArtifactHandler {
+
+  override suspend fun handle(artifact: DeliveryArtifact) {
```

Comment:
Do you have any ideas on how to make sure we don't keep rebaking forever? I just added this pr (https://github.com/spinnaker/keel/pull/929) to make sure we only try to rebake once. Moving baking out of the resource handler means that we can't take advantage of the `UnhappyVeto`. I'd really like that issue to not come back.